driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -145,6 +146,19 @@ public void initChannel(final SocketChannel ch) throws Exception {
             public void operationComplete(final ChannelFuture future) throws Exception {
                 if (future.isSuccess()) {
                     channel = channelFuture.channel();
+                    channel.closeFuture().addListener(new ChannelFutureListener() {
+                        @Override
+                        public void operationComplete(final ChannelFuture f2) throws Exception {
+                            PendingReader pending = null;
+                            synchronized (this) {
+                                pending = pendingReader;
+                                pendingReader = null;
+                            }
+                            if (pending != null) {
+                                pending.handler.failed(new MongoSocketException("The connection to the server was closed", address));
```

Comment:
Can just throw IOException as InternalStreamConnection#translateReadException will translate it properly.